apps/sms/services/js/messaging/messaging_service.js

```diff
@@ -65,6 +70,17 @@ var MessagingService = {
     this.initService();
   },
 
+  registerEvent(appInstanceId) {
+    var mobileMessageClient = MozMobileMessageClient.forApp(appInstanceId);
+
+    EVENTS.forEach((event) => {
+      mobileMessageClient.on(event, (...args) => {
+        this.broadcast(event, ...args);
```

Comment:
mm, I believe we don't need ```...``` here since we can have only one argument in the event handler usually, moreover  it won't work correctly in case of multiple arguments as 3rd argument of ```service.broadcast``` should be an array of clients to broadcast to :)

Speaking about third argument, looks like we need to revise our ```forApp``` and ```appInstanceId``` logic here - currently you broadcast to _all_ clients, so if we have two clients (app and activity) and we receive message, both iframes broadcast ```message-received``` and ```MessageService``` will get two events and .... broadcast them to both ```MessagingClient``` instances :) Likely we need ```Map<appInstanceId, bridgeClient>``` logic somewhere as well.

Also I see that ```alwaysLowered``` will not help here as well since both main app and iframe will have its own hidden window (need to talk to Francisco and Etienne to understand if it's expected that hidden window is tied to originating instance):(